Header header
float32 velocity
int32 counter